adapters/repos/db/vector/hnsw/neighbor_connections.go

```diff
@@ -321,11 +321,17 @@ func (n *neighborFinderConnector) connectNeighborAtLevel(neighborID uint64,
 
 	maximumConnections := n.maximumConnections(level)
 	if len(currentConnections) < maximumConnections {
-		// we can simply append
-		// updatedConnections = append(currentConnections, n.node.id)
-		neighbor.appendConnectionAtLevelNoLock(level, n.node.id, maximumConnections)
-		if err := n.graph.commitLog.AddLinkAtLevel(neighbor.id, level, n.node.id); err != nil {
-			return err
+		alreadyConnected := false
```

Comment:
couldn't this validation (alreadyConnected) be done even when len(currentConnections) < maximumConnections doesn't hold?

Reply:
It actually should. More than that, I added a second test to actively check if we have duplicates (even though the maxConnections is not violated) and we still had some. The second branch was adding just in rare occasions but still. Also, by adding this test, it is not flaky failing anymore. It consistently fails before the fix now.